src/io_scheduler.cpp

```diff
@@ -183,7 +183,7 @@ auto io_scheduler::poll(fd_t fd, coro::poll_op op, std::chrono::milliseconds tim
 auto io_scheduler::shutdown() noexcept -> void
 {
     // Only allow shutdown to occur once.
-    if (m_shutdown_requested.exchange(true, std::memory_order::acq_rel) == false)
+    if (m_shutdown_requested.exchange(true, std::memory_order::seq_cst) == false)
```

Comment:
I think we can back these out since the `notify_all()` was the problem.  Can you revert all of these seq_cst changes back to what they were please?